ably.ts.d

```diff
@@ -0,0 +1,457 @@
+  }
+  type ConnectionEvent = ConnectionEvent.INITIALIZED | ConnectionEvent.CONNECTED | ConnectionEvent.CONNECTING | ConnectionEvent.DISCONNECTED | ConnectionEvent.SUSPENDED | ConnectionEvent.CLOSED | ConnectionEvent.CLOSING | ConnectionEvent.FAILED | ConnectionEvent.UPDATE;
+
+  namespace PresenceAction {
+    export type ABSENT = "absent";
+    export type PRESENT = "present";
+    export type ENTER = "enter";
+    export type LEAVE = "leave";
+    export type UPDATE = "update";
+  }
+  type PresenceAction = PresenceAction.ABSENT | PresenceAction.PRESENT | PresenceAction.ENTER | PresenceAction.LEAVE | PresenceAction.UPDATE;
+
+  namespace StatsIntervalGranularity {
+    export type MINUTE = "minute";
+    export type HOUR = "hour";
+    export type DAY = "day";
+    export type MONTH = "month";
+  }
+  type StatsIntervalGranularity = StatsIntervalGranularity.MINUTE | StatsIntervalGranularity.HOUR | StatsIntervalGranularity.DAY | StatsIntervalGranularity.MONTH;
+
+  namespace ProtocolMessageAction {
+    export type HEARTBEAT = "heartbeat";
+    export type ACK = "ack";
+    export type NACK = "nack";
+    export type CONNECT = "connect";
+    export type CONNECTED = "connected";
+    export type DISCONNECT = "disconnect";
+    export type DISCONNECTED = "disconnected";
+    export type CLOSE = "close";
+    export type CLOSED = "closed";
+    export type ERROR = "error";
+    export type ATTACH = "attach";
+    export type ATTACHED = "attached";
+    export type DETACH = "detach";
+    export type DETACHED = "detached";
+    export type PRESENCE = "presence";
+    export type MESSAGE = "message";
+    export type SYNC = "sync";
+    export type AUTH = "auth";
+  }
+  type ProtocolMessageAction = ProtocolMessageAction.ACK | ProtocolMessageAction.ATTACH | ProtocolMessageAction.ATTACHED | ProtocolMessageAction.AUTH | ProtocolMessageAction.CLOSE | ProtocolMessageAction.CLOSED | ProtocolMessageAction.CONNECT | ProtocolMessageAction.CONNECTED | ProtocolMessageAction.DETACH | ProtocolMessageAction.DETACHED | ProtocolMessageAction.DISCONNECT | ProtocolMessageAction.DISCONNECTED | ProtocolMessageAction.ERROR | ProtocolMessageAction.HEARTBEAT | ProtocolMessageAction.MESSAGE | ProtocolMessageAction.NACK | ProtocolMessageAction.PRESENCE | ProtocolMessageAction.SYNC;
+
+
+  // Interfaces
+  interface AuthDetails {
+    accessToken: String;
+  }
+
+  interface ClientOptions extends AuthOptions {
+    autoConnect?: boolean;
+    clientId?: String;
+    defaultTokenParams?: TokenParams;
+    echoMessages?: boolean;
+    environment?: String;
+    logHandler?: any;
+    logLevel?: any;
+    port?: number;
+    queueMessages?: boolean;
+    restHost?: String;
+    realtimeHost?: String;
+    fallbackHosts?: Array<String>
+    recover?: String
+    tls?: boolean;
+    tlsPort?: number;
+    useBinaryProtocol?: boolean;
+    disconnectedRetryTimeout?: number;
+    suspendedRetryTimeout?: number;
+    channelRetryTimeout?: number;
+    httpOpenTimeout?: number;
```

Comment:
In the IDL, means it was translated.  :grinning:.   But I've dropped them from the next version of the typigns.  Thanks.